src/syntax/ppx_sedlex.ml

```diff
@@ -253,24 +264,208 @@ let gen_recflag auto =
     Nonrecursive
   with Exit -> Recursive
 
+let gen_cset ~loc x char_set =
+  let interval a b =
+    [%expr
+      [%e eint ~loc a] <= [%e evar ~loc x]
+      && [%e evar ~loc x] <= [%e eint ~loc b]]
+  in
+  match char_set with
+    | (a, b) :: l ->
+        List.fold_left
+          (fun acc (a, b) -> [%expr [%e acc] || [%e interval a b]])
+          (interval a b) l
+    | [] -> assert false
+
+let gen_trace lexbuf traces i = function
+  | (_, []), _ -> []
+  | (_, aliases), _ ->
+      let loc = default_loc in
+      let initial, trans, finals = traces.(i) in
+      let alias_indexes =
+        List.to_seq aliases
+        |> Seq.mapi (fun i { txt = alias } -> (alias, i))
+        |> StringMap.of_seq
+      in
+      let find_index alias = StringMap.find alias alias_indexes in
+      let result_arrays =
+        [
+          value_binding ~loc
+            ~pat:[%pat? __sedlex_aliases_starts]
```

Comment:
unifying `__sedlex_aliases_starts` and `__sedlex_aliases_stops` into a single `__sedlex_aliases_pos` could be a first step toward sharing more positions between aliases.

Reply:
This has been done.